frc46_token/src/token/state.rs

```diff
@@ -506,114 +511,169 @@ impl TokenState {
     /// Checks that there are no zero balances, zero allowances or empty allowance maps explicitly
     /// stored in the blockstore. Checks that balances, total supply, allowances are never negative.
     /// Checks that sum of all balances matches total_supply. Checks that no allowances are stored
-    /// where operator == owner.
+    /// where operator == owner. Checks that all balances are a multiple of the granularity.
+    ///
+    /// Returns a state summary that can be used to check application specific invariants.
     pub fn check_invariants<BS: Blockstore>(
         &self,
         bs: &BS,
-    ) -> std::result::Result<(), StateInvariantError> {
+        granularity: u64,
+    ) -> std::result::Result<StateSummary, Vec<StateInvariantError>> {
```

Comment:
Can you drop the Result and just return a tuple? We still want the state summary even if there are problems. Goal is to find all the errors in the whole system in a single run (though we can't get all the way there of course).